static/js/components/dashboard/CourseDescription.js

```diff
@@ -68,13 +68,34 @@ export default class CourseDescription extends React.Component {
     return _.compact([dateMessage, additionalDetail]);
   }
 
-  renderViewCourseLink = (courseRun: CourseRun): React$Element<*>|null => (
-    (courseRun && courseRun.course_id) ?
-      <a href={`${EDX_LINK_BASE}${courseRun.course_id}`} target="_blank">
-        View on edX
-      </a> :
-      null
-  );
+  isCurrentOrPastEnrolled = (courseRun: CourseRun): boolean => {
+    if([STATUS_CURRENTLY_ENROLLED, STATUS_PASSED, STATUS_NOT_PASSED].includes(courseRun.status)) {
+      return true;
+    } else {
+      if ([STATUS_CAN_UPGRADE, STATUS_MISSED_DEADLINE].includes(courseRun.status)) {
+        let now = moment();
+        let courseStart = moment(courseRun.course_start_date);
+        return courseStart.isBefore(now);
```

Comment:
this logic needs to change. if `courseRun.course_start_date` is null or undefined, `moment(courseRun.course_start_date)` will just behave like `moment()` and return the current datetime. this would work: 

```javascript
let now = moment();
return courseRun.course_start_date && moment(courseRun.course_start_date).isBefore(now);
```